helm/cluster/values.schema.json

```diff
@@ -1186,6 +1189,24 @@
                                 },
                                 "preKubeadmCommands": {
                                     "$ref": "#/$defs/preKubeadmCommands"
+                                },
+                                "rolloutBefore": {
+                                    "type": "object",
+                                    "title": "Rollout Before",
+                                    "description": "Rollout Before is a field to indicate a rollout should be performed if the specified criteria is met.",
+                                    "required": [
+                                        "certificatesExpiryDays"
+                                    ],
+                                    "properties": {
+                                        "certificatesExpiryDays": {
+                                            "type": "integer",
+                                            "title": "Certificate expiry days",
+                                            "description": "Indicates a rollout needs to be performed if the certificates of the machine will expire within the specified days.",
+                                            "default": 180,
+                                            "maximum": 364,
```

Comment:
Certificates in CAPI clusters expire after a year by default. If the `certificatesExpiryDays` value is set to 365 days before the certificate expires, then the nodes are rolled continuously and the rollout never stops. The maximum value of 364 days ensures that certificates are not rotated for at least a day.

In theory it is possible to set an expiration period shorter or longer than 1 year. In that case the maximum value can be restrictive. If the certificate expiration period is set to 2 years, then the maximum value will make it impossible to trigger the rollout more than 1 year before the certificates expire. 

Question is if there is a need for such use case - certificates valid for 2 years rotated more than a year before they expire.

Reply:
IMO this here is more than good enough for now.